lib/voight_kampff/test.rb

```diff
@@ -1,7 +1,48 @@
+# frozen_string_literal: true
+
 module VoightKampff
+  # Test User-Agent against Voight-Kampff
   class Test
     CRAWLERS_FILENAME = 'crawler-user-agents.json'
 
+    class << self
+      def crawlers
+        @crawlers ||= JSON.parse File.read preferred_path
+      end
+
+      def crawler_regexp
+        @crawler_regexp ||= begin
+          # NOTE: This is admittedly a bit convoluted
+          # but the performance gains make it worthwhile
+          crawler_patterns =
+            crawlers.map.with_index do |crawler, index|
+              "(?<match#{index}>#{crawler['pattern']})"
+            end.join('|')
+          crawler_patterns = "(#{crawler_patterns})"
+          Regexp.new(crawler_patterns, Regexp::IGNORECASE)
+        end
+      end
+
+      def lookup_paths
+        # These paths should be orderd by priority
+        [
+          (Rails.root if defined? Rails),
+          Dir.pwd,
+          VoightKampff::ROOT
```

Comment:
The constant ROOT is needed only here, may be calculate root path of the gem here without unnecessary constants?

Reply:
I've made it similar to `Rails.root` at least, I yes, it's using only in single place right now, but I also think this approach is more safe if the `test.rb` file will be renamed or just a method will be moved to another file. I see no problem with this simple constant. Again: it's more mnemonic to Rails and maybe even `Dir.pwd`, also it adds readability (without additional `..`s and `File` and `()`).